cachi2/core/package_managers/pip.py

```diff
@@ -1765,47 +1795,45 @@ def _process_package_distributions(
             else:
                 log.info("Filtering out %s due to checksum mismatch", wheel.filename)
 
-        if len(filtered) > 0 and len(wheels) == 0:
+        if len(filtered) == 0 and len(sdists) == 0:
+            raise PackageRejected(
+                f"No archives found for package {name}=={version}",
+                solution=(
+                    "It seems that this version does not exist or isn't published.\n"
+                    "You may be able to specify the dependency directly via a URL instead, "
+                    "for example the wheel for a GitHub release."
+                ),
+                docs=PIP_NO_SDIST_DOC,
+            )
+        elif len(filtered) == 0:
+            log.warning("No wheel distributions found for package %s==%s", name, version)
+        elif len(wheels) == 0:
             log.warning("All %s wheel distributions were filtered out", name)
```

Comment:
What is `filtered` and what is `wheels`? From the log messages, it looks like `filtered` is the unfiltered list of wheels and `wheels` is the filtered list? :sweat_smile: